src/main/java/htsjdk/samtools/cram/CRAIIndex.java

```diff
@@ -55,68 +52,28 @@ public void writeIndex(final OutputStream os) {
 
     /**
      * Create index entries for a single container.
-     * @param c the container to index
+     * @param container the container to index
      */
-    public void processContainer(final Container c) {
+    public void processContainer(final Container container) {
         // TODO: this should be refactored and delegate to container/slice
-        if (!c.isEOF()) {
-            for (int i = 0; i < c.slices.length; i++) {
-                Slice s = c.slices[i];
-                if (s.sequenceId == Slice.MULTI_REFERENCE) {
-                    this.entries.addAll(getCRAIEntriesForMultiRefSlice(s, c.header, c.offset, c.landmarks));
-                }
-                else {
-                    CRAIEntry e = new CRAIEntry();
-
-                    e.sequenceId = c.sequenceId;
-                    e.alignmentStart = s.alignmentStart;
-                    e.alignmentSpan = s.alignmentSpan;
-                    e.containerStartOffset = c.offset;
-                    e.sliceOffset = c.landmarks[i];
-                    e.sliceSize = s.size;
-                    e.sliceIndex = i;
-
-                    entries.add(e);
+        if (!container.isEOF()) {
+            for (final IndexableSlice slice : container.slices) {
+                if (slice.hasMultipleReferences()) {
+                    final Map<Integer, AlignmentSpan> spans = slice.getMultiRefAlignmentSpans(container.header, ValidationStringency.DEFAULT_STRINGENCY);
+                    final List<CRAIEntry> entries = spans
+                            .entrySet()
+                            .stream()
+                            .map(span -> slice.getCRAIEntry(span.getKey(), span.getValue(), container.offset))
+                            .collect(Collectors.toList());
+                    this.entries.addAll(entries);
+                } else {
+                    this.entries.add(slice.getCRAIEntry(container.offset));
                 }
             }
         }
     }
 
-    /**
-     * Return a list of CRAI Entries; one for each reference in the multireference slice.
-     * TODO: this should be refactored and delegate to container/slice
-     */
-    private static Collection<CRAIEntry> getCRAIEntriesForMultiRefSlice(
```

Comment:
inlined above, with the `CRAIEntry` creation step changed to a stream-mapping of `Slice.getCRAIEntry()`